adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -182,6 +182,7 @@ func (h *hnsw) resetUnlocked() error {
 	h.currentMaximumLayer = 0
 	h.initialInsertOnce = &sync.Once{}
 	h.nodes = make([]*vertex, cache.InitialSize)
+	h.tombstones = make(map[uint64]struct{})
```

Comment:
In other code paths `tombstones` are protected by `tombstoneLock`, however, `resetIfEmpty` and `resetIfOnlyNode` does not take this look, should we add it?

Reply:
Definitely, I think it is enough with the following change?